dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/enrollment/EnrollmentOperationParamsMapper.java

```diff
@@ -0,0 +1,188 @@
+import org.hisp.dhis.trackedentity.TrackedEntityType;
+import org.hisp.dhis.trackedentity.TrackedEntityTypeService;
+import org.hisp.dhis.trackedentity.TrackerAccessManager;
+import org.hisp.dhis.user.CurrentUserService;
+import org.hisp.dhis.user.User;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Transactional;
+
+/**
+ * Maps operation parameters stored in {@link EnrollmentOperationParams} to
+ * {@link EnrollmentQueryParams} which is used to fetch enrollments from the DB.
+ */
+@Component
+@RequiredArgsConstructor
+class EnrollmentOperationParamsMapper
+{
+    private final CurrentUserService currentUserService;
+
+    private final OrganisationUnitService organisationUnitService;
+
+    private final ProgramService programService;
+
+    private final TrackedEntityTypeService trackedEntityTypeService;
+
+    private final TrackedEntityService trackedEntityService;
+
+    private final TrackerAccessManager trackerAccessManager;
+
+    @Transactional( readOnly = true )
+    public EnrollmentQueryParams map( EnrollmentOperationParams operationParams )
+        throws BadRequestException,
+        ForbiddenException
+    {
+        Program program = validateProgram( operationParams.getProgramUid() );
+        TrackedEntityType trackedEntityType = validateTrackedEntityType( operationParams.getTrackedEntityTypeUid() );
+        TrackedEntity trackedEntity = validateTrackedEntity( operationParams.getTrackedEntityUid() );
+
+        User user = currentUserService.getCurrentUser();
+        Set<OrganisationUnit> orgUnits = validateOrgUnits( user, operationParams.getOrganisationUnitUids(), program );
+
+        EnrollmentQueryParams params = new EnrollmentQueryParams();
+        params.setProgram( program );
+        params.setProgramStatus( operationParams.getProgramStatus() );
+        params.setFollowUp( operationParams.getFollowUp() );
+        params.setLastUpdated( operationParams.getLastUpdated() );
+        params.setLastUpdatedDuration( operationParams.getLastUpdatedDuration() );
+        params.setProgramStartDate( operationParams.getProgramStartDate() );
+        params.setProgramEndDate( operationParams.getProgramEndDate() );
+        params.setTrackedEntityType( trackedEntityType );
+        params.setTrackedEntity( trackedEntity );
+        params.addOrganisationUnits( orgUnits );
+        params.setOrganisationUnitMode( operationParams.getOrganisationUnitMode() );
+        params.setPage( operationParams.getPage() );
+        params.setPageSize( operationParams.getPageSize() );
+        params.setTotalPages( operationParams.isTotalPages() );
+        params.setSkipPaging( toBooleanDefaultIfNull( operationParams.isSkipPaging(), false ) );
+        params.setIncludeDeleted( operationParams.isIncludeDeleted() );
+        params.setUser( user );
+        params.setOrder( operationParams.getOrder() );
+
+        return params;
+    }
+
+    private Program validateProgram( String uid )
```

Comment:
will these four validations remain in this mapper?

Reply:
For now I would say yes. If then we feel the need to extract them to a validator, we can go that way.